atuin-client/src/database.rs

```diff
@@ -242,7 +242,11 @@ impl Database for Sqlite {
             FilterMode::Directory => format!("cwd = '{}'", context.cwd).to_string(),
         };
 
-        let filter = format!("{} {}", join, filter_query);
+        let filter = if filter_query.is_empty() {
+            "".to_string()
```

Comment:
you are very lucky that rust is smart enough to avoid a new allocation for these excessive `.to_string()`s

Reply:
I thought that if the compiler _wasn't_ smart enough to do that, then I'd be deeply disappointed in Rust